app/jobs/gallery_validation_job.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+##
+# Validate that a gallery has only properly displaying images
+# Should an issue be discovered, a slack channel will be notified by sending an email.
+class GalleryValidationJob < ApplicationJob
+  queue_as :default
+
+  def perform(gallery_id)
+    @gallery = Gallery.find(gallery_id)
+    @validation_errors = {}
+    @gallery.images.each do |gallery_image|
+      validate_gallery_image(gallery_image)
+    end
+    notify if @validation_errors.count.positive?
+  end
+
+  private
+
+  def validate_gallery_image(gallery_image)
+    errors = []
+    europeana_id = gallery_image.europeana_record_id
+    stored_image_url = gallery_image.image_url
+    api_document = api_search_response.detect { |record| record['id'] == europeana_id }
+    if api_document
+      unless api_document['edmIsShownBy'] == stored_image_url
+        errors << "edm:isShownBy for image '#{europeana_id}' has changed"
+      end
+      unless retrievable_image?(stored_image_url)
+        errors << "The image '#{europeana_id}' can not be retrieved at '#{stored_image_url}'."
```

Comment:
This should differentiate between non-retrievable URLs (404, 500, etc) and retrievable URLs which are not images.

Reply:
Does this really matter to an editor?